### ISSUE TITLE
Resolve Node.js warnings (dependencies)

### ISSUE THREAD

**Issue by DanielRuf** [magento/magento2#21403]
### Preconditions (*)

1. in magento root/ rename package.json.example to  package.json
2. npm install

### Steps to reproduce (*)

1. Install the depencies with Yarn / npm
2. npm audit

### Expected result (*)

1. [No outdated / insecure dependencies.

### Actual result (*)

```
warning grunt-autoprefixer > autoprefixer-core > browserslist@0.4.0: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
warning grunt-contrib-connect > http2@3.3.7: Use the built-in module in node 9.0.0 or newer, instead
warning grunt-contrib-imagemin > imagemin > imagemin-jpegtran > jpegtran-bin > bin-build > download > gulp-decompress > gulp-util@3.0.8: gulp-util is deprecated - replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
warning grunt-contrib-less > less > request > hawk > cryptiles@2.0.5: This version is no longer maintained. Please upgrade to the latest version.
warning grunt-contrib-less > less > request > hawk > hoek@2.16.3: This version is no longer maintained. Please upgrade to the latest version.
warning grunt-contrib-less > less > request > hawk > sntp > hoek@2.16.3: This version is no longer maintained. Please upgrade to the latest version.
warning grunt-contrib-less > less > request > hawk > boom@2.10.1: This version is no longer maintained. Please upgrade to the latest version.
warning grunt-contrib-less > less > request > hawk > cryptiles > boom@2.10.1: This version is no longer maintained. Please upgrade to the latest version.
warning grunt-contrib-less > less > request > hawk > boom > hoek@2.16.3: This version is no longer maintained. Please upgrade to the latest version.
warning grunt-eslint > eslint > file-entry-cache > flat-cache > circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
warning grunt-jscs > jscs > jscs-preset-wikimedia@1.0.1: Deprecated in favour of eslint-config-wikimedia. -- https://phabricator.wikimedia.org/T118941
warning grunt-jscs > jscs > jsonlint > nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
warning grunt-replace > applause > cson-parser > coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
warning grunt-styledocco > styledocco > jade@0.23.0: Jade has been renamed to pug, please install the latest version of pug instead of jade
warning load-grunt-config > cson > coffee-script@1.12.7: CoffeeScript on NPM has moved to "coffeescript" (no hyphen)
warning node-minify > babel-preset-babili@0.0.12: babili has been renamed to babel-minify. Please update to babel-preset-minify
```

```
yarn audit
...
60 vulnerabilities found - Packages audited: 10367
Severity: 33 Low | 23 Moderate | 4 High
```

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@DanielRuf do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
HI @DanielRuf thank you for you report, Can someone solve it here? I'm afraid that this ticket may hang here for a long time

**Comment by DanielRuf** [comment]
@magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by DanielRuf** [comment]
@magento-engcom-team give me 2.2.7 instance

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf. Thank you for your request. I'm working on Magento 2.2.7 instance for you

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf, here is your [Magento instance](https://i-21403-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-21403-2-2-develop.instances.magento-community.engineering/admin](https://i-21403-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98382`, `MAGETWO-98383` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24319` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
